packages/agreement-process/src/services/readModelService.ts

```diff
@@ -562,6 +611,69 @@ export function readModelServiceBuilder(
         ),
       };
     },
+    async getDelegationByDelegateId(
+      delegateId: TenantId,
+      kind: DelegationKind,
+      state: DelegationState = delegationState.active
+    ): Promise<WithMetadata<Delegation> | undefined> {
+      const data = await delegations.findOne(
+        {
+          "data.delegateId": delegateId,
+          "data.state": state,
+          "data.kind": kind,
+        },
+        { projection: { data: true, metadata: true } }
+      );
+
+      if (!data) {
+        return undefined;
+      }
+      const result = z
+        .object({
+          data: Delegation,
+          metadata: z.object({ version: z.number() }),
+        })
+        .safeParse(data);
+      if (!result.success) {
+        throw genericInternalError(
+          `Unable to parse delegation item: result ${JSON.stringify(
+            result
+          )} - data ${JSON.stringify(data)} `
+        );
+      }
+      return result.data;
+    },
+    async getActiveDelegationByEserviceId(
+      eserviceId: EServiceId,
+      kind: DelegationKind
+    ): Promise<WithMetadata<Delegation> | undefined> {
+      const data = await delegations.findOne(
+        {
+          "data.eserviceId": eserviceId,
+          "data.state": delegationState.active,
+          "data.kind": kind,
+        },
+        { projection: { data: true, metadata: true } }
```

Comment:
```suggestion
    ): Promise<Delegation | undefined> {
      const data = await delegations.findOne(
        {
          "data.eserviceId": eserviceId,
          "data.state": delegationState.active,
          "data.kind": kind,
        },
        { projection: { data: true } }
```
We don't need the metadata here too

Reply:
Absolutely make sense! 👍🏻 
removed in [2776c23](https://github.com/pagopa/interop-be-monorepo/pull/1175/commits/2776c23a311157fe4b184123300b15545c8c9d24)